iidm/iidm-api/src/main/java/com/powsybl/iidm/network/limitreduction/computer/AbstractReducedLimitsComputer.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.network.limitreduction.computer;
+
+import com.powsybl.iidm.network.LimitType;
+import com.powsybl.iidm.network.ThreeSides;
+import com.powsybl.iidm.network.limitreduction.result.LimitsContainer;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * <p>Abstract class responsible for computing reduced limits.</p>
+ * <p>Already computed reduced limits are stored in an internal cache to avoid unnecessary computations.
+ * This cache should be cleared with {@link #clearCache()} when the reductions to apply are changed.</p>
+ *
+ * @author Olivier Perrin {@literal <olivier.perrin at rte-france.com>}
+ */
+public abstract class AbstractReducedLimitsComputer<P, L> implements ReducedLimitsComputer<P, L> {
```

Comment:
We have an interface called `LimitReduction`. I think that a good way to have an affordable code is to re-use this naming in your utility method.